### PR TITLE
Puppet e2e

### PR DESCRIPTION
**Pull request by Gauravtalreja1**
1. Mark e2e in Puppet tests and some CaseImportance changes
2. Moved test_positive_install_configure to test_reports

### CODE REVIEW

File: tests/foreman/cli/test_report.py [hunk 1]
```diff
@@ -22,6 +22,8 @@
 
 from robottelo.cli.base import CLIReturnCodeError
 
+pytestmark = pytest.mark.e2e
```
Comment (Griffin-Sullivan) [inline tests/foreman/cli/test_report.py:25]: Could you educate me on what this does?

**Review by Griffin-Sullivan — APPROVED** [review]
Looks good! Just a few questions and would like to learn a little.

File: tests/foreman/api/test_classparameters.py [hunk 2]
```diff
@@ -78,6 +80,7 @@ def module_puppet(session_puppet_enabled_sat):
 
 
 @pytest.mark.run_in_one_thread
+@pytest.mark.e2e
```
[the remark this answers is not in the document]
Comment (Griffin-Sullivan) [reply tests/foreman/api/test_classparameters.py:83]: My main concern would be test bloat. I feel like it is important for our e2e suite to be really the bare minimum.

[the remark this answers is not in the document]
Comment (shweta83) [reply tests/foreman/api/test_classparameters.py:83]: I agree.We can mark individual tests.

File: tests/foreman/cli/test_report.py [hunk 3]
```diff
@@ -68,3 +66,46 @@ def test_positive_delete_by_id(run_puppet_agent, session_puppet_enabled_sat):
     session_puppet_enabled_sat.cli.ConfigReport.delete({'id': report['id']})
     with pytest.raises(CLIReturnCodeError):
         session_puppet_enabled_sat.cli.ConfigReport.info({'id': report['id']})
+
+
+@pytest.mark.e2e
+@pytest.mark.rhel_ver_match('[7,8]')
+def test_positive_install_configure(request, session_puppet_enabled_sat, rhel_contenthost):
```
Comment (shweta83) [inline tests/foreman/cli/test_report.py:73]: Why this test case is moved out from destructive module?

File: tests/foreman/api/test_environment.py [hunk 4]
```diff
@@ -30,6 +30,14 @@
 from robottelo.utils.datafactory import valid_environments_list
 
 
+@pytest.fixture(scope='module')
+def module_locations(session_puppet_enabled_sat):
```
[the remark this answers is not in the document]
Comment (jameerpathan111) [reply tests/foreman/api/test_environment.py:34]: Nope, at least not in `tests/foreman/api/test_environment.py`, seems like he added it by mistake.

File: tests/foreman/api/test_classparameters.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Gauravtalreja1) [reply tests/foreman/api/test_classparameters.py:83]: @Griffin-Sullivan @shweta83 Yes, it should be the bare minimum but yet effective, I've got these tests from the old component owner, so I think we should keep this as it is, and then we can later revisit this, sounds good?

File: tests/foreman/api/test_environment.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Gauravtalreja1) [reply tests/foreman/api/test_environment.py:34]: Yes, we can remove this

File: tests/foreman/cli/test_report.py (hunk 1 above)
Comment (Gauravtalreja1) [reply tests/foreman/cli/test_report.py:25]: @Griffin-Sullivan Sure, it means all tests under this module will be marked as e2e

File: tests/foreman/cli/test_report.py (hunk 3 above)
Comment (Gauravtalreja1) [reply tests/foreman/cli/test_report.py:73]: @shweta83 this test is not at all destructive, so moved this is out of destructive

File: tests/foreman/cli/test_report.py (hunk 1 above)
Comment (Griffin-Sullivan) [reply tests/foreman/cli/test_report.py:25]: Seen it before and was assuming that's what happens but never knew for sure. Thanks!

**Review by adarshdubey-star — APPROVED** [review]

File: tests/foreman/api/test_environment.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Gauravtalreja1) [reply tests/foreman/api/test_environment.py]: removed

**Review by jameerpathan111 — APPROVED** [review]

**Pull request merged**